src/Search_Replace_Command.php

```diff
@@ -197,6 +200,12 @@ public function __invoke( $args, $assoc_args ) {
 				}
 				WP_CLI::error( $msg );
 			}
+			if ( ( $regex_limit = (int) \WP_CLI\Utils\get_flag_value( $assoc_args, 'regex-limit', - 1 ) ) > - 1 ) {
```

Comment:
This check should be moved to the previous block where the other options are checked (after line 183) and should not allow zero as it doesn't make sense to me. So I'd prefer:
```php
			if ( null !== ( $regex_limit = \WP_CLI\Utils\get_flag_value( $assoc_args, 'regex-limit' ) ) ) {
				if ( ! preg_match( '/^(?:[0-9]+|-1)$/', $regex_limit ) || 0 === (int) $regex_limit ) {
					WP_CLI::error( '`--regex-limit` expects a non-zero positive integer or -1.' );
				}
				$this->regex_limit = (int) $regex_limit;
			}
```